openmls/src/treesync/diff.rs

```diff
@@ -171,29 +178,39 @@ impl<'a> TreeSyncDiff<'a> {
         &mut self,
         leaf_node: KeyPackage,
         backend: &impl OpenMlsCrypto,
-    ) -> Result<LeafIndex, TreeSyncDiffError> {
-        let node = Node::LeafNode(LeafNode::new(leaf_node, backend).map_err(|e| {
-            if let KeyPackageError::CryptoError(e) = e {
-                TreeSyncDiffError::CryptoError(e)
-            } else {
-                TreeSyncDiffError::LibraryError(LibraryError::custom("key package error"))
-            }
-        })?);
+    ) -> Result<LeafIndex, TreeSyncAddLeaf> {
+        let node = Node::LeafNode(LeafNode::new(leaf_node, backend)?);
         // Find a free leaf and fill it with the new key package.
         let leaf_index = self.free_leaf_index()?;
         // If the free leaf index is within the tree, put the new leaf there,
         // otherwise extend the tree.
         if leaf_index < self.leaf_count() {
-            self.diff.replace_leaf(leaf_index, node.into())?;
+            self.diff
+                .replace_leaf(leaf_index, node.into())
+                .map_err(|_| LibraryError::custom("no parent hash extension"))?;
```

Comment:
I'm confused. Why would an error thrown by `replace_leaf` indicate that a parent hash extension is missing?

Unrelared nit: In most other cases, you added a nice note on why we're mapping to a `LibraryError`. Would you mind adding that here as well?